src/Ipns.ts

```diff
@@ -0,0 +1,49 @@
+import fetchPonyfill from 'fetch-ponyfill'
+
+import { CheckBase } from './CheckBase'
+import { IPNS_PATH_TO_TEST } from './constants'
+import type { GatewayNode } from './GatewayNode'
+
+import { Log } from './Log'
+
+const { fetch } = fetchPonyfill()
+
+const log = new Log('Ipns')
+
+class IPNSCheck extends CheckBase implements Checkable {
+  _className = 'Ipns'
+  _tagName = 'div'
+  constructor (protected parent: GatewayNode) {
+    super(parent, 'div', 'Ipns')
+  }
+
+  async check (): Promise<void> {
+    const now = Date.now()
+    const gatewayUrl = new URL(this.parent.gateway)
+    gatewayUrl.pathname = IPNS_PATH_TO_TEST
+    const testUrl = `${gatewayUrl.href}?now=${now}`
+    try {
+      const response = await fetch(testUrl)
+      if (response.status === 200) {
+        this.tag.win()
+      } else {
+        log.debug(`${this.parent.gateway} does not support IPNS`)
+        throw new Error(`URL '${testUrl} is not reachable`)
+      }
+    } catch (err) {
+      log.error(err)
+      this.onerror()
+      throw err
+    }
+  }
```

Comment:
I imagine we would also get some timeouts from gateways that would produce false-negative results when trying to retrieve larger data; Something smaller should reduce that risk.

Reply:
> Is there a more lightweight method of accomplishing this? e.g. inline IPNS image? 

Not such thing, I am afraid. We could have IPNS set up just for testing, but it requires infra which I don't want to introduce to this project. 

Wikipedia Mirror is future-proof, as it will survive everything else, and is stable.
- `/ipns/en.wikipedia-on-ipfs.org/favicon.ico`  is pretty future-proof, most likely will always be present, and image is only 4960 bytes
- We could generate smaller load by requesting `/ipns/en.wikipedia-on-ipfs.org/index.html` which is a redirect to `/wiki` that is only 169 bytes atm, but that can change in the future.

I'd keep favicon. This is nothing compared to real world abuse Gateways see :upside_down_face:

Reply:
@lidel I agree, but there is also: https://ipfs.io/ipns/en.wikipedia-on-ipfs.org/I/1x1.png.webp if we wanna be cautious.

Reply:
This is not a stable URL that is guaranteed to work when a new snapshot is published.

Reply:
I'll leave it at `favicon.ico` till something better comes along